plugins/utils/toggleterm.nix

```diff
@@ -0,0 +1,160 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.toggleterm;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.toggleterm = {
+    enable = mkEnableOption "toggleterm";
+
+    package = helpers.mkPackageOption "toggleterm" pkgs.vimPlugins.toggleterm-nvim;
+
+    size =
+      helpers.defaultNullOpts.mkNullable
+      (with types; either number str) "" ''
+        size can be a number or function which is passed the current terminal
+        Example:
+        size = 20
+        OR
+        size = function(term)
+          if term.direction == "horizontal" then
+            return 15
+          elseif term.direction == "vertical" then
+            return vim.o.columns * 0.4
+          end
+        end
+      '';
+
+    openMapping = helpers.defaultNullOpts.mkStr "[[<c-\\>]]" "Setting the open_mapping key to use for toggling the terminal(s) will set up mappings for normal mode";
```

Comment:
Sorry for being picky but I will ask you to start sentences with a capital letter and end them with a period ('.').
This is so that we keep consistency across documentation :)